scholarx/archive/functions.js

```diff
@@ -52,13 +58,121 @@ async function getData() {
-    $("#mentorProfiles").html(mentor_Profiles);
+    const { data } = await getData();
+    mentors = data.mentors;
+    mentees = data.mentees;
+    mentorProfilesData = mentors[2019].concat(mentors[2020]).concat(mentors[2021]);
+    let mentorProfiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorProfilesData });
+    $("#mentorProfiles").html(mentorProfiles);
 
-    const menteeProfiles = payload.mentee2021.concat(payload.mentee2020).concat(payload.mentee2019);
-    let mentee_Profiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeProfiles });
-    $("#menteeProfiles").html(mentee_Profiles);
+    menteeProfilesData = mentees[2019].concat(mentees[2020]).concat(mentees[2021]);
+    let menteeProfiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeProfilesData });
+    $("#menteeProfiles").html(menteeProfiles);
 }
 loadData();
+function renderProfiles(mentorYear,menteeYear) {
+    let mentorProfiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorYear });
+    let menteeProfiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeYear });
+    $("#mentorProfiles").html(mentorProfiles);
+    $("#menteeProfiles").html(menteeProfiles);
+}
+function renderAllProfiles() {
+    let mentorProfiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorProfilesData });
+    let menteeProfiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeProfilesData });
+    $("#mentorProfiles").html(mentorProfiles);
+    $("#menteeProfiles").html(menteeProfiles);
+}
+//function to return true if all industry checkboxes are unchecked.
+function allFalse(){
+    for(let i=1; i<7; i++){
+        if(document.getElementById("industry_"+i).checked == true){
+            return false
+        }
+    }return true
+}
+//function to uncheck checkboxes
+function uncheckFilters(){
+    let year = 2019
+    for(let i=1; i<7; i++){
+        document.getElementById("industry_"+i).checked = false;
+    }
+    for(let i=0; i<4; i++){
+        document.getElementById(year).checked = false
+        year++
+    }
+}
+//year by filters
+function filterByYear(){
+    let year = 2019;
+    let mentorsData = [];
+    let menteesData = [];
+    if(document.getElementById("2019").checked == false && document.getElementById("2020").checked == false && document.getElementById("2021").checked == false){
+        renderAllProfiles();
+    }else{
+        for(let i=0; i<3; i++){
+            if(document.getElementById(year).checked)
+            {
+                mentorsData = mentorsData.concat(mentors[year]);
+                menteesData = menteesData.concat(mentees[year]);
+                dataHolder = mentorsData; 
+                if(allFalse()){
+                    renderProfiles(mentorsData,menteesData)
+                }else{
+                    filterByIndustry("value")
```

Comment:
Why do we pass a "value" here?

Reply:
> Why do we pass a "value" here?

It isn't required but we can put something other than "all"